### ISSUE TITLE
EXCEPTION: System.ObjectDisposedException: Cannot access a disposed object.

### ISSUE THREAD

**Issue by OscarMinassian** [bloomberg/collectdwin#43]
Hello Friends,

Let me start by saying, awesome product. Collectdwin is allowing us to completely remove our old monitoring platform and replace it entirely with a monster Splunk cluster.

We've deployed the Bloomberg client out to our entire fleet of Windows servers and in about 95% of cases it's humming along just fine and dandy. 

The remaining 5% though is having some exceptions that I'm struggling to get around. I suspect that there might be some crossover with other products that are running on these servers but I'm really unsure as to what and they varied service. eg Veeam backup servers, TeamCity, Octopus Deploy.

Any assistance in this would be HUGELY appreciated. 

The exception that's seen is as follows:
`2018-03-08 13:39:43.5981|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: WindowsPerformanceCounter reloading configuration
2018-03-08 13:40:13.6789|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: WindowsPerformanceCounter reloading configuration
2018-03-08 13:40:43.7600|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: WindowsPerformanceCounter reloading configuration
2018-03-08 13:40:55.0367|INFO|BloombergFLP.CollectdWin.StatsdPlugin: Statsd plugin stopped
2018-03-08 13:40:55.0367|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: WindowsPerformanceCounter plugin stopped
2018-03-08 13:40:55.0367|INFO|BloombergFLP.CollectdWin.ConsolePlugin: console plugin stopped
2018-03-08 13:40:55.0367|ERROR|BloombergFLP.CollectdWin.MetricsCollector: WriteThreadProc() got exception : 
EXCEPTION: System.Threading.ThreadInterruptedException: Thread was interrupted from a waiting state.
   at System.Threading.Thread.SleepInternal(Int32 millisecondsTimeout)
   at System.Threading.Thread.Sleep(Int32 millisecondsTimeout)
   at BloombergFLP.CollectdWin.MetricsCollector.WriteThreadProc()
2018-03-08 13:40:55.0367|ERROR|BloombergFLP.CollectdWin.Program: CollectdWin: exiting ...
2018-03-08 13:40:55.0367|ERROR|BloombergFLP.CollectdWin.MetricsCollector: ReadThreadProc() got exception : System.Threading.ThreadInterruptedException: Thread was interrupted from a waiting state.
   at System.Threading.Thread.SleepInternal(Int32 millisecondsTimeout)
   at System.Threading.Thread.Sleep(Int32 millisecondsTimeout)
   at BloombergFLP.CollectdWin.MetricsCollector.ReadThreadProc(),    at System.Threading.Thread.SleepInternal(Int32 millisecondsTimeout)
   at System.Threading.Thread.Sleep(Int32 millisecondsTimeout)
   at BloombergFLP.CollectdWin.MetricsCollector.ReadThreadProc()

2018-03-08 13:40:55.0367|ERROR|BloombergFLP.CollectdWin.MetricsCollector: AggregatorThreadProc() got exception : 
EXCEPTION: System.Threading.ThreadInterruptedException: Thread was interrupted from a waiting state.
   at System.Threading.Thread.SleepInternal(Int32 millisecondsTimeout)
   at System.Threading.Thread.Sleep(Int32 millisecondsTimeout)
   at BloombergFLP.CollectdWin.MetricsCollector.AggregatorThreadProc()
2018-03-08 13:40:55.0367|ERROR|BloombergFLP.CollectdWin.StatsdListener: CloseSocket failed: 
EXCEPTION: System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'System.Net.Sockets.Socket'.
   at System.Net.Sockets.Socket.Shutdown(SocketShutdown how)
   at BloombergFLP.CollectdWin.StatsdListener.CloseSocket()
2018-03-08 13:41:37.7290|INFO|BloombergFLP.CollectdWin.StatsdAggregator: Statsd config - delCounter:True, delTimer:True, delGauge:True, delSet:True, isHistogramEnabled:True
2018-03-08 13:41:37.7580|INFO|BloombergFLP.CollectdWin.StatsdPlugin: Statsd plugin configured
2018-03-08 13:41:37.7730|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Processor', Name = '% Processor Time', Instance = '_Total', CollectdPlugin = 'cpu', CollectdPluginInstance = '', CollectdType = 'percent', CollectdTypeInstance = 'processor', 
2018-03-08 13:41:38.8283|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Processor - Instance:_Total - counter:% Processor Time
2018-03-08 13:41:38.8283|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Processor', Name = '% Idle Time', Instance = '_Total', CollectdPlugin = 'cpu', CollectdPluginInstance = '', CollectdType = 'percent', CollectdTypeInstance = 'idle', 
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Processor - Instance:_Total - counter:% Idle Time
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Processor', Name = '% Privileged Time', Instance = '_Total', CollectdPlugin = 'cpu', CollectdPluginInstance = '', CollectdType = 'percent', CollectdTypeInstance = 'system', 
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Processor - Instance:_Total - counter:% Privileged Time
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Processor', Name = '% User Time', Instance = '_Total', CollectdPlugin = 'cpu', CollectdPluginInstance = '', CollectdType = 'percent', CollectdTypeInstance = 'user', 
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Processor - Instance:_Total - counter:% User Time
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Processor', Name = '% Interrupt Time', Instance = '_Total', CollectdPlugin = 'cpu', CollectdPluginInstance = '', CollectdType = 'percent', CollectdTypeInstance = 'interrupt', 
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Processor - Instance:_Total - counter:% Interrupt Time
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Memory', Name = 'Available Bytes', Instance = '', CollectdPlugin = 'memory', CollectdPluginInstance = '', CollectdType = 'memory', CollectdTypeInstance = 'free', 
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Memory - Instance: - counter:Available Bytes
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = 'MemoryAvailableBytesToUsedPercentage', TransformerParameters = '', Category = 'Memory', Name = 'Available Bytes', Instance = '', CollectdPlugin = 'memory', CollectdPluginInstance = '', CollectdType = 'percent', CollectdTypeInstance = 'used', 
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Cannot find method for creating metric generator: Transformer=MemoryAvailableBytesToUsedPercentage, using default generator.
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Memory - Instance: - counter:Available Bytes
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Memory', Name = 'Page Faults/sec', Instance = '', CollectdPlugin = 'memory', CollectdPluginInstance = '', CollectdType = 'memory', CollectdTypeInstance = 'faults_per_sec', 
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Memory - Instance: - counter:Page Faults/sec
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Memory', Name = '% Committed Bytes In Use', Instance = '', CollectdPlugin = 'memory', CollectdPluginInstance = '', CollectdType = 'memory', CollectdTypeInstance = 'percent_committed_inuse', 
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Memory - Instance: - counter:% Committed Bytes In Use
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Paging File', Name = '% Usage', Instance = '_Total', CollectdPlugin = 'percent', CollectdPluginInstance = '', CollectdType = 'swap', CollectdTypeInstance = 'percent_used', 
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Paging File - Instance:_Total - counter:% Usage
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = 'Averages', TransformerParameters = '60,300,900', Category = 'System', Name = 'Processor Queue Length', Instance = '', CollectdPlugin = 'load', CollectdPluginInstance = '', CollectdType = 'load', CollectdTypeInstance = '', 
2018-03-08 13:41:38.8293|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:System - Instance: - counter:Processor Queue Length
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'System', Name = 'System Up Time', Instance = '', CollectdPlugin = 'uptime', CollectdPluginInstance = '', CollectdType = 'uptime', CollectdTypeInstance = '', 
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:System - Instance: - counter:System Up Time
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'System', Name = 'System Calls/sec', Instance = '', CollectdPlugin = 'system', CollectdPluginInstance = '', CollectdType = 'gauge', CollectdTypeInstance = 'sys_calls_per_sec', 
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:System - Instance: - counter:System Calls/sec
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'System', Name = 'File Data Operations/sec', Instance = '', CollectdPlugin = 'io', CollectdPluginInstance = '', CollectdType = 'gauge', CollectdTypeInstance = 'file_ops_per_sec', 
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:System - Instance: - counter:File Data Operations/sec
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'System', Name = 'File Read Operations/sec', Instance = '', CollectdPlugin = 'io', CollectdPluginInstance = '', CollectdType = 'gauge', CollectdTypeInstance = 'file_read_ops_per_sec', 
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:System - Instance: - counter:File Read Operations/sec
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'System', Name = 'File Write Operations/sec', Instance = '', CollectdPlugin = 'io', CollectdPluginInstance = '', CollectdType = 'gauge', CollectdTypeInstance = 'file_write_ops_per_sec', 
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:System - Instance: - counter:File Write Operations/sec
2018-03-08 13:41:38.8583|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'LogicalDisk', Name = '% Free Space', Instance = '*', CollectdPlugin = 'df', CollectdPluginInstance = '', CollectdType = 'percent_bytes', CollectdTypeInstance = 'free', 
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:HarddiskVolume2 - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:I: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:H: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:E: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:N: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:G: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:_Total - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:C: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:F: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = 'PercentageToRemainingPercentage', TransformerParameters = '', Category = 'LogicalDisk', Name = '% Free Space', Instance = '*', CollectdPlugin = 'df', CollectdPluginInstance = '', CollectdType = 'percent_bytes', CollectdTypeInstance = 'used', 
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Cannot find method for creating metric generator: Transformer=PercentageToRemainingPercentage, using default generator.
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:HarddiskVolume2 - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:I: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:H: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:E: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:N: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:G: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:_Total - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:C: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:F: - counter:% Free Space
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = 'MegaBytesToBytes', TransformerParameters = '', Category = 'LogicalDisk', Name = 'Free Megabytes', Instance = '*', CollectdPlugin = 'df', CollectdPluginInstance = '', CollectdType = 'df_complex', CollectdTypeInstance = 'free', 
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Cannot find method for creating metric generator: Transformer=MegaBytesToBytes, using default generator.
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:HarddiskVolume2 - counter:Free Megabytes
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:I: - counter:Free Megabytes
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:H: - counter:Free Megabytes
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:E: - counter:Free Megabytes
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:N: - counter:Free Megabytes
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:G: - counter:Free Megabytes
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:_Total - counter:Free Megabytes
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:C: - counter:Free Megabytes
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:LogicalDisk - Instance:F: - counter:Free Megabytes
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = 'CountProcessesAndThreads', TransformerParameters = '', Category = 'Process', Name = '', Instance = '', CollectdPlugin = 'processes', CollectdPluginInstance = 'all', CollectdType = 'ps_count', CollectdTypeInstance = '', 
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Process', Name = 'Handle Count', Instance = '_Total', CollectdPlugin = 'processes', CollectdPluginInstance = '', CollectdType = 'gauge', CollectdTypeInstance = 'handles', 
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Process - Instance:_Total - counter:Handle Count
2018-03-08 13:41:38.8663|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Network Interface', Name = 'Packets Received/Sec,Packets Sent/Sec', Instance = '*', CollectdPlugin = 'interface', CollectdPluginInstance = '', CollectdType = 'if_packets', CollectdTypeInstance = '', 
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Network Interface - Instance:isatap.{5B788871-00F4-41F9-8E6E-5BEE7388CDB7} - counter:Packets Received/Sec,Packets Sent/Sec
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Network Interface - Instance:HP Ethernet 10Gb 2-port 530FLR-SFP+ Adapter  _74 - counter:Packets Received/Sec,Packets Sent/Sec
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Network Interface - Instance:HP Ethernet 10Gb 2-port 530FLR-SFP+ Adapter  _73 - counter:Packets Received/Sec,Packets Sent/Sec
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Network Interface', Name = 'Bytes Received/Sec,Bytes Sent/Sec', Instance = '*', CollectdPlugin = 'interface', CollectdPluginInstance = '', CollectdType = 'if_octets', CollectdTypeInstance = '', 
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Network Interface - Instance:isatap.{5B788871-00F4-41F9-8E6E-5BEE7388CDB7} - counter:Bytes Received/Sec,Bytes Sent/Sec
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Network Interface - Instance:HP Ethernet 10Gb 2-port 530FLR-SFP+ Adapter  _74 - counter:Bytes Received/Sec,Bytes Sent/Sec
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Network Interface - Instance:HP Ethernet 10Gb 2-port 530FLR-SFP+ Adapter  _73 - counter:Bytes Received/Sec,Bytes Sent/Sec
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: Configuring metric generator: Transformer = '', TransformerParameters = '', Category = 'Network Interface', Name = 'Packets Received Errors,Packets Outbound Errors', Instance = '*', CollectdPlugin = 'interface', CollectdPluginInstance = '', CollectdType = 'if_errors', CollectdTypeInstance = '', 
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Network Interface - Instance:isatap.{5B788871-00F4-41F9-8E6E-5BEE7388CDB7} - counter:Packets Received Errors,Packets Outbound Errors
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Network Interface - Instance:HP Ethernet 10Gb 2-port 530FLR-SFP+ Adapter  _74 - counter:Packets Received Errors,Packets Outbound Errors
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.PerformanceCounterGenerator: Added Performance COUNTER : Category:Network Interface - Instance:HP Ethernet 10Gb 2-port 530FLR-SFP+ Adapter  _73 - counter:Packets Received Errors,Packets Outbound Errors
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: WindowsPerformanceCounter plugin configured
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.ConsolePlugin: console plugin configured
2018-03-08 13:41:38.8763|INFO|BloombergFLP.CollectdWin.StatsdPlugin: Statsd plugin started
2018-03-08 13:41:38.8913|INFO|BloombergFLP.CollectdWin.WindowsPerformanceCounterPlugin: WindowsPerformanceCounter plugin started
2018-03-08 13:41:38.8913|INFO|BloombergFLP.CollectdWin.ConsolePlugin: console plugin started`

**Comment by yogeswaran** [comment]
ThreadInterruptedException happens when the collectdwin windows service is stopped, it looks like the service stopped and started again. Is it happening a lot or sometimes?

**Comment by OscarMinassian** [comment]
Hey @yogeswaran, 

Thanks for getting back to me, what you're seeing is me restarting the service to create the exception and attempt to get it sending data. It all seems to hum along fine but I'm not getting any output come through to my Splunk clusters.

I thought it might be something to do with the bellow. Apologies in advance for my blatant noobism :)

2018-03-08 13:40:55.0367|ERROR|BloombergFLP.CollectdWin.StatsdListener: CloseSocket failed:
EXCEPTION: System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'System.Net.Sockets.Socket'.
at System.Net.Sockets.Socket.Shutdown(SocketShutdown how)
at BloombergFLP.CollectdWin.StatsdListener.CloseSocket()

**Comment by yogeswaran** [comment]
ok, above exception is from [statsd plugin](https://github.com/bloomberg/collectdwin/wiki/StatsD-plugin#statsd-plugin), this plugin listens on a UDP port 8125. If it fails to bind to port, you would expect above exception.

I suspect either the service does not have permission to bind to that port or some other app is listening on that port. If you are not using statsd plugin, you can disable statsd plugin in the [plugin registry](https://github.com/bloomberg/collectdwin/wiki/Getting-started#plugin-registry). If you need it, then check if the port is already used by some other app or if collectdwin service has permission to bind to that port.

**Comment by OscarMinassian** [comment]
I ran a netstat -a -b and have found the CollectdWinService bound to 8125

  UDP    0.0.0.0:8125           *:*
 [CollectdWinService.exe]

Despite, I've disabled the statsd plugin and restarted the service. Doesn't seem like it's made any difference. Looking at the CollectdWin_debug.log it did get rid of the socket error but I'm still not getting anything passed through :|

I can see that the above binding to port 8125 has been removed since. Any ideas @yogeswaran ?
